pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/filter/TimePredicateFilterOptimizer.java

```diff
@@ -411,6 +428,80 @@ && isStringLiteral(dateTimeConvertOperands.get(3)),
     }
   }
 
+  private void optimizeDateTrunc(Function filterFunction, FilterKind filterKind) {
+    List<Expression> filterOperands = filterFunction.getOperands();
+    List<Expression> dateTruncOperands = filterOperands.get(0).getFunctionCall().getOperands();
+
+    // Check if date trunc function is being applied on a literal value
+    if (dateTruncOperands.get(1).isSetLiteral()) {
```

Comment:
Since `DATETRUNC` can be applied on literals, my thinking was that if we returned at this point, some other optimizer would precompute this value. I need to look into the other optimizers to figure out if this is the case. If this is not applied elsewhere, I'll add the computation for the `DATETRUNC` of the literal and new query creation here.